ldms/src/core/ldms_xprt.c

```diff
@@ -3922,7 +3928,6 @@ int __ldms_remote_lookup(ldms_t _x, const char *path,
 				LDMS_CONTEXT_LOOKUP_REQ,
 				cb, arg, lu_path, flags);
 	if (!ctxt) {
-		free(lu_path);
```

Comment:
@baallan, Ii think this change is incorrect. 

The lu_path is a strdup() of the path argument. This is passed to __ldms_remote_lookup() which takes this pointer and stuffs it in the ctxt structure.

So far, so good, however, if the calloc() of the ctxt structure fails, the lu_path is left unused and leaks the lu_path allocation. This is why the if (!ctxt) path frees lu_path.

This change also falls under the category of "if this fails the process is dying anyway because we're out of memory. In other words, it doesn't matter, other than quieting the tool.

That said, and out of curiosity, what tool gave you the advice that this needed to be removed, because I  believe that advice is incorrect.

Reply:
@tom95858 reviewing this on in the tool, i had marked it false positive because the tool didn't decode (because of the varargs) that lu_path was stored conditionally in the alloc function.  So i must have accidentally deleted a line here.
will put it back.